xc7/utils/prjxray_assign_tile_pin_direction.py

```diff
@@ -5,17 +5,202 @@
+
+        # Get the node that is attached to the sink.
+        c2.execute("""SELECT node_pkey FROM wire WHERE pkey = ?""",
+                (dest_wire_pkey,))
+        (dest_node_pkey,) = c2.fetchone()
+
+        # Find the wire connected to the sink.
+        dest_wire = list(node_to_site_pins(conn, dest_node_pkey))
+        assert len(dest_wire) == 1
+        destination_wire_pkey, dest_tile_pkey, dest_wire_in_tile_pkey = dest_wire[0]
+
+        c2.execute("""
+            SELECT tile_type_pkey, grid_x, grid_y FROM tile WHERE pkey = ?;""",
+                (dest_tile_pkey,))
+        dest_tile_type_pkey, destination_loc_grid_x, destination_loc_grid_y = c2.fetchone()
+
+        c2.execute("""SELECT name FROM tile_type WHERE pkey = ?""",
+                (dest_tile_type_pkey,))
+        (destination_tile_type,) = c2.fetchone()
+
+        c2.execute("""SELECT name FROM wire_in_tile WHERE pkey = ?""",
+                (dest_wire_in_tile_pkey,))
+        (destination_wire,) = c2.fetchone()
+
+        direct_connections.add(
+                DirectConnection(
+                        from_pin='{}.{}'.format(source_tile_type, source_wire),
+                        to_pin='{}.{}'.format(destination_tile_type, destination_wire),
+                        switch_name='routing',
+                        x_offset=destination_loc_grid_x-source_loc_grid_x,
+                        y_offset=destination_loc_grid_y-source_loc_grid_y,
+                )
+        )
+
+        if destination_loc_grid_x == source_loc_grid_x:
+            if destination_loc_grid_y > source_loc_grid_y:
+                source_dir = tracks.Direction.TOP
+                destination_dir = tracks.Direction.BOTTOM
+            else:
+                source_dir = tracks.Direction.BOTTOM
+                destination_dir = tracks.Direction.TOP
+        else:
+            if destination_loc_grid_x > source_loc_grid_x:
+                source_dir = tracks.Direction.RIGHT
+                destination_dir = tracks.Direction.LEFT
+            else:
+                source_dir = tracks.Direction.LEFT
+                destination_dir = tracks.Direction.RIGHT
+
+        edge_assignments[(source_tile_type, source_wire)].append((source_dir,))
+        edge_assignments[(destination_tile_type, destination_wire)].append((destination_dir,))
+
+
+def handle_edges_to_channels(conn, null_tile_wires, edge_assignments, channel_wires_to_tracks):
+    c = conn.cursor()
+
+    for node_pkey, classification in progressbar.progressbar(c.execute("""
+SELECT pkey, classification FROM node WHERE classification != ?;
+""", (NodeClassification.CHANNEL.value,))):
+        reason = NodeClassification(classification)
+
+        if reason == NodeClassification.NULL:
+            for (_, tile_type, wire) in yield_wire_info_from_node(conn, node_pkey):
+                null_tile_wires.add((tile_type, wire))
+
+        if reason != NodeClassification.EDGES_TO_CHANNEL:
+            continue
+
+        c2 = conn.cursor()
```

Comment:
I would say the SQL queries here are inconsistently formatted.

Reply:
Took another pass, take a look.